installer/cluster.go

```diff
@@ -0,0 +1,410 @@
+package installer
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"text/template"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/crypto/ssh"
+	cfg "github.com/flynn/flynn/cli/config"
+	"github.com/flynn/flynn/pkg/etcdcluster"
+)
+
+func (c *Cluster) saveField(field string, value interface{}) error {
+	c.installer.dbMtx.Lock()
+	defer c.installer.dbMtx.Unlock()
+
+	var err error
+
+	tx, err := c.installer.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	switch field {
+	case "InstanceIPs":
+		insertStmt := ""
+		fields := make([]interface{}, 0, len(c.InstanceIPs)+1)
+		fields = append(fields, c.ID)
+		for i, ip := range c.InstanceIPs {
+			insertStmt += fmt.Sprintf("INSERT INTO instances (ClusterID, IP) VALUES ($1, $%d);\n", i+2)
+			fields = append(fields, ip)
+		}
+		q := fmt.Sprintf(`
+    %s
+    `, insertStmt)
+		_, err = tx.Exec(q, fields...)
+	case "Domain":
```

Comment:
This is weird, I think it's better served by separate methods to do `Domain` and `InstanceIPs`.